blog/2019-03-14-debezium-meets-quarkus.adoc

```diff
@@ -0,0 +1,193 @@
+All that's needed to consume messages from a Kafka topic is to annotate a method with `@Incoming`,
+and will automatically be invoked when a new message arrives:
+
+[source,java]
+----
+@ApplicationScoped
+public class KafkaEventConsumer {
+
+    @Incoming("orders")
+    public CompletionStage<Void> onMessage(KafkaMessage<String, String> message)
+            throws IOException {
+        // handle message...
+
+        return message.ack();
+    }
+}
+----
+
+The "orders" message source is configured via the MicroProfile Config API,
+which resolves it to the "OrderEvents" topic already known from the original outbox example.
+
+== Build Process
+
+The build process is almost the same as it was before.
+Besides adding the Quarkus Maven plug-in itself, only two changes were needed:
+
+ * A new build profile named `native` has been added; this is used to compile the service into a native binary image using the Quarkus Maven plug-in
+ * the `native-image.docker-build` system property is enabled when running the build; this means that the native image build is done inside of a Docker container, so that GraalVM doesn't have to be installed on the developer's machine
+
+All the heavy-lifting is done by the Quarkus Maven plug-in which is configured in _pom.xml_ like so:
+
+[source,xml]
+----
+  <build>
+    <finalName>shipment</finalName>
+    <plugins>
+      ...
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-maven-plugin</artifactId>
+        <version>${version.quarkus}</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>build</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+  ...
+    <profile>
+      <id>native</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-maven-plugin</artifactId>
+            <version>${version.quarkus}</version>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>native-image</goal>
+                </goals>
+                <configuration>
+                  <enableHttpUrlHandler>true</enableHttpUrlHandler>
+                  <autoServiceLoaderRegistration>false</autoServiceLoaderRegistration>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+----
+
+== Configuration
+
+The application contains two configuration files:
+
+ * _application.properties_: the default configuration file of Quarkus applications; contains the definition of a datasource (based on MariaDB) to which the shipment service writes its data
+ * _META-INF/microprofile-config.properties_: contains a definition of the Kafka topic from which we consume the outbox events and the Kafka consumer group id
```

Comment:
Do you really need this second file?

I'm not sure we did all the work to use our config everywhere but that would be some interesting feedback.

Reply:
@jpechane, have you tried putting everything into _application.properties_? If it doesn't work, could you file an issue with Quarkus, as the file should be used as a source for MP Config, it might only be it's not propagated to Reactive Messaging yet. Thanks!

Reply:
@gunnarmorling
https://github.com/quarkusio/quarkus-quickstarts/issues/98
https://github.com/debezium/debezium-examples/pull/56

Reply:
@gsmet, so turns out just _application.properties_ is fine. The demo is already adjusted, I'll update the blog post PR next.